tests/streaming/test_streaming.py

```diff
@@ -249,11 +249,11 @@ def __call__(self, x: str, **kwargs):
             dspy.streaming.StreamListener(signature_field_name="judgement"),
         ],
         include_final_prediction_in_output_stream=False,
+        async_streaming=False,
```

Comment:
Is this test run in the CI pipeline? It seems to require `OPENAI_API_KEY`

Reply:
it's not running in CI, but written for local checking. streaming behavior is a bit unreliable to verify with mock, and we should find a way to set up API key for github actions.

Reply:
I see, is it difficult to mock the LiteLLM call with a dummy generator?

Reply:
Mocking streaming is messy but feasible, the more concerning part is the assumption on stream tokens can go wrong, which could lead to false positive test. But I agree with you that we can add another test based on mock. Merging this one now, will follow up with the test shortly.